cpp/tests/groupby/group_count_scan_test.cpp

```diff
@@ -0,0 +1,213 @@
+}
+
+TYPED_TEST(groupby_count_scan_test, zero_valid_values)
+{
+  using K = int32_t;
+  using V = TypeParam;
+  using R = cudf::detail::target_type_t<V, aggregation::COUNT_ALL>;
+
+  // clang-format off
+  fixed_width_column_wrapper<K> keys            {1, 1, 1};
+  fixed_width_column_wrapper<V, int> vals (     {3, 4, 5}, all_null());
+
+  fixed_width_column_wrapper<K> expect_keys     {1, 1, 1};
+  fixed_width_column_wrapper<R, int> expect_vals{0, 1, 2};
+  // clang-format on
+
+  auto agg2 = cudf::make_count_aggregation(null_policy::INCLUDE);
+  test_single_scan(keys, vals, expect_keys, expect_vals, std::move(agg2));
+}
+
+TYPED_TEST(groupby_count_scan_test, null_keys_and_values)
+{
+  using K = int32_t;
+  using V = TypeParam;
+  using R = cudf::detail::target_type_t<V, aggregation::COUNT_ALL>;
+
+  // clang-format off
+  fixed_width_column_wrapper<K> keys(     { 1, 2, 3, 1, 2, 2, 1, 3, 3, 2, 4},
+                                          { 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1});
+  fixed_width_column_wrapper<V, int> vals({ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 4},
+                                          { 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 0});
+
+                                        //  {1, 1, 1, 2, 2, 2, 2, 3, _, 3, 4}
+  fixed_width_column_wrapper<K> expect_keys({1, 1, 1, 2, 2, 2, 2, 3,    3, 4}, all_valid());
+                                        //  {0, 3, 6, 1, 4, _, 9, 2, 7, 8, -}
+  fixed_width_column_wrapper<R, int> expect_vals {0, 1, 2, 0, 1, 2, 3, 0, 1, 0};
+  // clang-format on
+
+  auto agg2 = cudf::make_count_aggregation(null_policy::INCLUDE);
+  test_single_scan(keys, vals, expect_keys, expect_vals, std::move(agg2));
+}
+
+struct groupby_count_scan_string_test : public cudf::test::BaseFixture {
+};
+
+TEST_F(groupby_count_scan_string_test, basic)
+{
+  using K = int32_t;
+  using V = cudf::string_view;
+  using R = cudf::detail::target_type_t<V, aggregation::COUNT_ALL>;
+
+  // clang-format off
+  fixed_width_column_wrapper<K> keys        {   1,   3,   3,   5,   5,   0};
+  strings_column_wrapper        vals        { "1", "1", "1", "1", "1", "1"};
+
+  fixed_width_column_wrapper<K> expect_keys     {0, 1, 3, 3, 5, 5};
+  fixed_width_column_wrapper<R, int> expect_vals{0, 0, 0, 1, 0, 1};
+  // clang-format on
+
+  auto agg2 = cudf::make_count_aggregation(null_policy::INCLUDE);
+  test_single_scan(keys, vals, expect_keys, expect_vals, std::move(agg2));
+}
+
+template <typename T>
+struct FixedPointTestBothReps : public cudf::test::BaseFixture {
+};
+
+TYPED_TEST_CASE(FixedPointTestBothReps, cudf::test::FixedPointTypes);
+
+TYPED_TEST(FixedPointTestBothReps, GroupByCountScan)
+{
+  using namespace numeric;
+  using decimalXX  = TypeParam;
+  using RepType    = cudf::device_storage_type_t<decimalXX>;
+  using fp_wrapper = cudf::test::fixed_point_column_wrapper<RepType>;
+
+  using K = int32_t;
+  using V = decimalXX;
+  using R = cudf::detail::target_type_t<V, aggregation::COUNT_ALL>;
```

Comment:
These lines appear too many times. As suggested earlier, you can use the define macro to alias them at the beginning of the file.